### PR TITLE
external-apps: Support decompressing data.tar.xz for deb assets

### PR DESCRIPTION
**Pull request by joaquimrocha**
https://phabricator.endlessm.com/T11731

### CODE REVIEW

File: src/plugins/gs-plugin-external-apps.c [hunk 1]
```diff
@@ -269,18 +269,45 @@ add_runtime_deb_asset (const gchar *build_dir,
 		       GError **error)
 {
 	const char *argv[] = {"ar", "x",  asset_path, NULL};
-	const char *argv2[] = {"tar", "xzf",  "data.tar.gz", "-C", repo_dir,
-			       NULL};
+	const char **argv2 = NULL;
```
Comment (cosimoc) [inline src/plugins/gs-plugin-external-apps.c]: Should not be `const` if you allocate it later

File: src/plugins/gs-plugin-external-apps.c [hunk 2]
```diff
@@ -269,18 +269,45 @@ add_runtime_deb_asset (const gchar *build_dir,
...
+	g_autofree char *data_tar_path = NULL;
+	gboolean ret = FALSE;
 
 	if (!run_command (build_dir, argv, error)) {
 		return FALSE;
 	}
 
-	if (!run_command (build_dir, argv2, error)) {
-		return FALSE;
+	data_tar_path = g_strconcat (build_dir, "/data.tar.gz", NULL);
```
Comment (cosimoc) [inline src/plugins/gs-plugin-external-apps.c]: Use `g_build_filename()`

File: src/plugins/gs-plugin-external-apps.c (hunk 1 above)
Comment (cosimoc) [inline src/plugins/gs-plugin-external-apps.c]: You can also make it `g_autoptr(GStrv)` IIRC

File: src/plugins/gs-plugin-external-apps.c [hunk 3]
```diff
@@ -269,18 +269,45 @@ add_runtime_deb_asset (const gchar *build_dir,
...
+				     "package '%s' in '%s'", asset_path,
+				     build_dir);
+
+			return FALSE;
+		}
 	}
 
-	return TRUE;
+	/* Build array for command: tar xf TAR_FILE -C REPO_DIR */
+	argv2 = (const char **) g_new0(char *, 6);
```
Comment (cosimoc) [inline src/plugins/gs-plugin-external-apps.c]: Missing space before paren

File: src/plugins/gs-plugin-external-apps.c (hunk 1 above)
Comment (cosimoc) [inline src/plugins/gs-plugin-external-apps.c]: Ooh I see, only the outer array is allocated

Comment (joaquimrocha) [inline src/plugins/gs-plugin-external-apps.c]: Yeah. That's why I haven't used the autoptr :)

**Pull request merged**